Enable color output from integration tests in Docker containers

gtrafimenkov opened the pull request:
##### ISSUE TYPE
- Feature Pull Request
##### ANSIBLE VERSION

```
ansible 2.0.2.0
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

Enable color output from integration tests in Docker containers when then output is attached to the a terminal.  It is much easier to see results of the test this way.

gtrafimenkov committed:
Enable color output from integration tests in Docker containers

when then output is attached to a terminal.

It is much easier to see results of the test this way.

jctanner commented:
@mattclay does this look okay to you? I am in favor of merging.

gundalow commented:
@mattclay Would a similar patch also need applying to https://github.com/ansible/ansible/blob/devel/test/utils/shippable/integration.sh assuming Shippable’s test console handle termcolor?

mattclay commented:
@gtrafimenkov Sorry, I saw this earlier then forgot to comment. As @gundalow mentioned, it would be good to include this for shippable as well, since we're using it instead of Travis.

Would you mind trying to implement it there as well?

gtrafimenkov committed:
Enable colors in Shippable console log

That makes it much easier to read Ansible output.

gtrafimenkov commented:
Yes, Shippable's test console can handle colors.  I've made necessary changes in the `integration.sh` script.

It looks awesome: https://db.tt/sw4Pg8RA

mattclay commented on test/utils/shippable/integration.sh:
```diff
@@ -67,8 +67,12 @@ if [ "${controller_shared_dir}" ]; then
 fi
 
+# Enabling colored output in Shippable console
+color_settings="--env ANSIBLE_FORCE_COLOR=1"
```
Forcing color output causes causes the `blocks` integration test to fail. You can test this with:

`IMAGE=ansible/ansible:ubuntu1604 TARGET=blocks test/utils/shippable/integration.sh`

This is due to the test using grep on the playbook output. It probably wouldn't hurt to modify the test to work with or without color output.

I do like having the option to force color output, as it makes it possible to have color when running tests locally and saving them to a log file. I think the ideal solution would be to have force color the default, but being able to disable it when running integration.sh locally. You'll have to fix the tests to work with color though.

gtrafimenkov committed:
Make blocks tests work with or without color output

gtrafimenkov committed:
Can disable color output force mode

Color output is forced by default.

Buy it may be turned off with env variable FORCE_COLOR=0

gtrafimenkov commented:
@mattclay thank you for your feedback.  I've made the necessary changes.  All tests are passed now.

mattclay commented on test/utils/shippable/integration.sh:
```diff
@@ -67,8 +71,12 @@ if [ "${controller_shared_dir}" ]; then
+
 httptester_id=$(docker run -d "${http_image}")
 container_id=$(docker run -d \
+    ${color_settings} \
```
Inline `--env ANSIBLE_FORCE_COLOR=$force_color` here. Doing so avoids an extra variable and allows this script to pass shellcheck.

mattclay replied:
Correction, use `--env "ANSIBLE_FORCE_COLOR=${force_color}"`.

mattclay commented:
@gtrafimenkov This makes the build output so much easier to read, both locally and on shippable. Thanks for taking the time to work on this. This will be ready to merge once you make the change to allow shellcheck to pass on integration.sh (see my previous comment).

gtrafimenkov committed:
Avoid introducing extra variable

also pass shellcheck

gtrafimenkov commented:
@mattclay Thank you.  I am glad to help.

gundalow commented:
Excellent, thanks for this
